modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/fork/ForkInfoStorageSuite.scala

```diff
@@ -0,0 +1,90 @@
+package org.tessellation.sdk.infrastructure.fork
+
+import cats.effect.{IO, Ref}
+
+import scala.collection.immutable.Queue
+
+import org.tessellation.schema.generators.{peerIdGen, snapshotOrdinalGen}
+import org.tessellation.sdk.config.types.ForkInfoStorageConfig
+import org.tessellation.sdk.domain.fork.{ForkInfo, ForkInfoStorage, StoredForkInfo}
+import org.tessellation.security.hash.Hash
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.cats.refTypeShow
+import eu.timepit.refined.numeric.Positive
+import eu.timepit.refined.types.numeric.PosInt
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object ForkInfoStorageSuite extends SimpleIOSuite with Checkers {
+  val genStoredForkInfoEntry: Gen[StoredForkInfo] = for {
+    peerId <- peerIdGen
+    ordinal <- snapshotOrdinalGen
+    hashString <- Gen.alphaStr
+    hash = Hash(hashString)
+  } yield StoredForkInfo(peerId, ForkInfo(ordinal, hash))
+
+  def mkForkInfoStorage(queue: Queue[StoredForkInfo], queueSize: PosInt): F[ForkInfoStorage[ForkInfoStorageSuite.F]] = {
+    val config = ForkInfoStorageConfig(queueSize)
+
+    Ref[IO].of(queue).map(ForkInfoStorage.make(_, config))
+  }
+
+  test("initial store is empty") {
+    val gen = for {
+      queueSize <- Gen.chooseNum(2, 10)
+      refinedQueueSize = Refined.unsafeApply[Int, Positive](queueSize)
+    } yield refinedQueueSize
```

Comment:
Do you really need a `forall` for this? checking for empty queue is not dependent on queue size

Reply:
Replaced the generator with `Random.between`.